cuda_pathfinder/tests/child_load_nvidia_dynamic_lib_helper.py

```diff
@@ -50,4 +57,4 @@ def child_process_func(libname):
         raise RuntimeError(f"not os.path.samefile({loaded_dl_no_cache.abs_path=!r}, {loaded_dl_fresh.abs_path=!r})")
     validate_abs_path(loaded_dl_no_cache.abs_path)
 
-    sys.stdout.write(f"{loaded_dl_fresh.abs_path!r}\n")
+    sys.stdout.write(json.dumps(loaded_dl_fresh.abs_path) + "\n")
```

Comment:
Why are you using `sys.stdout.write` here instead of `print`, which by default appends the newline that you're explicitly writing here?

```suggestion
    print(json.dumps(loaded_dl_fresh.abs_path))
```

Reply:
It's either `sys.stdout.write` or `# noqa: T201`

```
rwgk-win11.localdomain:~/forked/cuda-python/cuda_pathfinder $ pre-commit run --all-files
ruff (legacy alias)......................................................Failed
- hook id: ruff
- exit code: 1

cuda_pathfinder/tests/child_load_nvidia_dynamic_lib_helper.py:60:5: T201 `print` found
   |
58 |     validate_abs_path(loaded_dl_no_cache.abs_path)
59 |
60 |     print(json.dumps(loaded_dl_fresh.abs_path))
   |     ^^^^^ T201
   |
   = help: Remove `print`

Found 1 error.
No fixes available (1 hidden fix can be enabled with the `--unsafe-fixes` option).
```

Generally I try to minimize the use of `noqa`.